src/cleanfigure.m

```diff
@@ -939,6 +939,30 @@ function removeNaNs(meta, handle)
     topRight    = [xLim(2); yLim(2)];
 end
 % =========================================================================
+function data = limitPrecision(data)
+  % Limit the precision of the given data
+  
+  % Only do something if the data is not empty
+  if isempty(data) || all(~isfinite(data))
+      return
+  end
+  
+  % Only consider finite values
+  mask = isfinite(data);
+
+  % Get the maximal value of the data
+  range = max(abs(data(mask)));
+
+  % Get the maximal precision for the data range
+  eps_range = eps(range);
+
+  % Scale the data wrt the range
+  data  = data / eps_range;
+
+  % Round to precision and scale back
+  data  = round(data) * eps_range;
+end
```

Comment:
I would put the precision reduction on a single line, i.e.
```matlab
% the least significant bit is proportional
% to the numerical precision of the largest number
lsb = eps(range) * alpha;

data = round(data/lsb) * lsb;
```
since that expresses best what the intent is. Putting everything across different lines is often nice, but here it's gone a bit overboard.

Note that, `alpha` is supposed to become a parameter to the `cleanfigure` function. Just imagine `plot(x, sin(x))`. The easiest to see the effect is around `0`: `sin(x)` may be small, e.g. `1e-14`, which in practice is a zero as one is likely to see. But `1e-14` will not be put to zero since `eps = 2e-16`. This `alpha` allows to put the bar a bit lower for this.